### ISSUE TITLE
Tags separate from note files

### ISSUE THREAD

**Issue by embamba** [notable/notable#767]
It would be nice if tags could be separate from the note file they're attached to. As it is now, if I tag a two year old file, the modification time stamp for that file changes to the moment in which the tag was applied. In other words, right now if I tag it now. But since I haven't edited the contents of the file, it's more valuable for me that the file remains listed in the file list way below, two years ago when it was last modified. So for my uses, I'd be better off tagging only current notes and ones I'd edit in the future. I'd group tag a bunch of notes if it didn't mean reordering their presence in the file list.

**Comment by fabiospampinato** [comment]
I see 3 possible options here:

1) Separated metadata, doesn't affect the `modified` date
2) Front matter metadata, `modified` is updated even if only the metadata changes
3) Front matter metadata, `modified` is updated only when the actual content of a note changes

I'm not sure what the best option here would be, perhaps this could be customizable.

Maybe we should switch to option 3 by default?

**Comment by embamba** [comment]
Does "Front matter metadata" refer to the metadata when it's at the head of the content of the text file that contains the note? If that's the case, I'd probably still vote for option 1, though option 3 would be an improvement over the current approach. I prefer option 1 because when I use other editors, on my phone, for example, those editors will show the metadata if it's in the file. For me, keeping a text file "clean" so that it only has content and no metadata is preferable.

Nice work on this. Thanks for responding

**Comment by fabiospampinato** [comment]
> Does "Front matter metadata" refer to the metadata when it's at the head of the content of the text file that contains the note?

Yes.

> I prefer option 1 because when I use other editors, on my phone, for example, those editors will show the metadata if it's in the file. For me, keeping a text file "clean" so that it only has content and no metadata is preferable.

They will show it in the rendered note? If so I think the issue here is that those Markdown editors should be updated with support for front matter.

I personally prefer option 3 to option 1, having the metadata embedded in the note means that you can edit a note's tags also pretty easily with an external editor, and finding notes by tag via grep becomes possible 🤔

**Comment by embamba** [comment]
> They will show it in the rendered note? If so I think the issue here is that those Markdown editors should be updated with support for front matter.

Good point. I hadn't realized that front matter was part of any Markdown standard. I checked some apps/programs I use that support Markdown. Some hide the front matter, others don't.

> I personally prefer option 3 to option 1, having the metadata embedded in the note means that you can edit a note's tags also pretty easily with an external editor, and finding notes by tag via grep becomes possible 🤔

Works for me.
